src/components/reference-editor/spec.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import test from 'tape';
+import { shallow, mount } from 'enzyme';
+import spy, { createSpy } from '../../utils/spy';
+
+import {
+  EditorState,
+  ContentState,
+  ContentBlock,
+  CharacterMetadata,
+  convertToRaw,
+  convertFromRaw,
+  genKey,
+} from 'draft-js';
+
+import EditorFactory, { UpstreamEditor, mentionPlugin, linkifyPlugin, undoPlugin } from './';
+const Editor = EditorFactory(React);
+
+test('ReferenceEditor', t => {
+  let instance, actual, expected;
+
+  const content = {};
+
+  instance = shallow(<Editor content={content} searchSuggestions={[]} />);
+
+  {
+    const upstream = instance.find(UpstreamEditor).at(0);
+    t.ok(upstream, 'should render the editor');
+  }
+
+  {
+    const upstream = instance.find(UpstreamEditor).at(0);
+    const plugins = upstream.props().plugins;
+    t.notEquals(plugins.indexOf(mentionPlugin), -1, 'should manage mention plugin');
```

Comment:
From looking at these tests, it seems the whole discussion about the "factory factory" might be related to just these three lines. If so, in the future, I don't feel this is something we would necessarily need to test. To some degree, it's just testing for a typo. If we wanted to be extra sure, we could check that the number of plugins passed matches our expectation, but I don't think we got a lot of value out of this particular set of assertions.

It's fine to leave, since the work is already done, but for the future, I consider stuff like this superfluous, so you need't worry about testing it too much.